### PR TITLE
enable jshint undef

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Our new JS pipeline branch has undef enabled, and I noticed it was hitting a bunch of errors - some of which seemed legit.

I decided to enable in our existing branch and fix all the errors. One of these revealed a legit bug (where we weren't properly animating Scrat's victory dance).

### CODE REVIEW

File: apps/src/maze/maze.js [hunk 1]
```diff
@@ -1601,8 +1601,8 @@ function setPegmanTransparent() {
  * @param {integer} timeAlloted How much time we have for our animations
  */
 function scheduleDance(victoryDance, timeAlloted) {
-  if (mazeUtils.isScratSkin()) {
-    scrat.scheduleDance(victoryDance, timeAlloted);
+  if (mazeUtils.isScratSkin(skin.id)) {
+    scrat.scheduleDance(victoryDance, timeAlloted, skin);
```
Comment (breville) [inline apps/src/maze/maze.js]: aha, thanks!

Comment (Bjvanminnen) [inline apps/src/maze/maze.js]: Np, it was my bug. I did a refactor at one point, apparently without properly testing the code. Jshint would have caught it if we had undef enabled ;)

File: apps/src/StudioApp.js [hunk 2]
```diff
@@ -1,5 +1,4 @@
-// Globals:
-//   Blockly
+/* global Blockly, ace:true, $, requirejs */
```
Comment (islemaster) [inline apps/src/StudioApp.js]: Is the `true` in `ace:true` needed here?

:+1: for jshint-annotating globals.

File: apps/src/StudioApp.js [hunk 3]
```diff
@@ -446,6 +445,7 @@ StudioApp.prototype.handleSharing_ = function (options) {
     }
     if (belowVisualization) {
       if (options.noButtonsBelowOnMobileShare) {
+        var visualization = document.getElementById('visualization');
         belowVisualization.style.display = 'none';
         visualization.style.marginBottom = '0px';
```
Comment (islemaster) [inline apps/src/StudioApp.js]: Was this even working before?  And if not, will this affect eyes tests?

**Comment by islemaster** [comment]
Looks great!  It will be great to have `undef` enabled, it's one of the style rules that regularly catches actual bugs.

File: apps/src/StudioApp.js (hunk 3 above)
Comment (Bjvanminnen) [inline apps/src/StudioApp.js]: It was working. apparently when we do container.innerHTML = foo, any HTML in foo that has an id gets loaded into the global namespace. This seems crazy to me, but it's why both this and runButton (the next change) worked before.

**Pull request merged**